nixos/modules/services/backup/duplicati.nix

```diff
@@ -0,0 +1,48 @@
+{ config, pkgs, lib, mono, ... }:
+
+with lib;
+
+let
+  cfg = config.services.duplicati;
+in
+{
+  options = {
+    services.duplicati = {
+      enable = mkEnableOption "Duplicati";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.duplicati = {
+      description = "Duplicati backup";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      preStart = ''
+        test -d /var/lib/duplicati/ || {
+          echo "Creating duplicati data directory in /var/lib/duplicati/"
+          mkdir -p /var/lib/duplicati/
+        }
+        chown -R duplicati:duplicati /var/lib/duplicati/
+        chmod 0700 /var/lib/duplicati/
+      '';
```

Comment:
You can add `createHome = true` to the user declaration, then all of this is done automatically for you and you can omit `preStart` and `PermissionsStartOnly`.